Avatar component

Monte9 opened the pull request:
Please review @binoy14 

TODO:
- [x] Update props to be deconstructed within the component itself and pass in ( ...props) outside.
- [x] Add ability to render initials too (either image or initials for avatar)
- [x] Update README to include example and docs for Avatar component

Example:
``` js
<Avatar
  small
  rounded
  icon={{type: 'home', size: 23}}
  onPress={() => console.log("Works!")}
  activeOpacity={0.7}
  containerStyle={{flex: 2, marginLeft: 20, marginTop: 115}}
/>
<Avatar
  medium
  source={{uri: "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"}}
  onPress={() => console.log("Works!")}
  activeOpacity={0.7}
  containerStyle={{flex: 3, marginTop: 100}}
/>
<Avatar
  large
  rounded
  title="MT"
  onPress={() => console.log("Works!")}
  activeOpacity={0.7}
  containerStyle={{flex: 4, marginTop: 75}}
/>
<Avatar
  xlarge
  rounded
  title="MT"
  source={{uri: "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"}}
  onPress={() => console.log("Works!")}
  activeOpacity={0.7}
  containerStyle={{flex: 5, marginRight: 60}}
/>
```

Here's what it looks like: 

<img width="445" alt="screen shot 2017-02-26 at 7 07 37 pm" src="https://cloud.githubusercontent.com/assets/7840686/23347531/cec23444-fc57-11e6-8518-3abcb14d1df0.png">

binoy14 commented on src/avatar/Avatar.js:
```diff
@@ -0,0 +1,131 @@
+      paddingTop: 10,
+      paddingRight: 10,
+      paddingBottom: 10,
+      borderBottomColor: '#ededed',
```
@Monte9 What is the reason for adding a borderBottom?

Monte9 replied:
That's unnecessary. I'll there caused I copy pasted the style from another component. 
I'll remove it.

binoy14 commented:
@Monte9 Few Issues I found.

1. The component is not exported in index.js
2. When the width is less than ~150 the title does not show up. It is probably because font is bigger and it's overflowing 
3. Based on the previous comment I am not sure what is the need for having a bottom border 
4. For destructuring you can use the following snippet.

```js
const Avatar = (props) => {
  const {
    component,
    onPress,
    onLongPress,
    containerStyle,
    source,
    avatarStyle,
    roundAvatar,
    title,
    titleStyle,
    overlayContainerStyle,
    activeOpacity,
  } = props;

// Because we are overriding width and height it can't be a const
  let {
    width,
    height,
  } = props;

  if (!width && !height) {
    width = 34
    height = 34
  } else if (!width) {
    width = height
  } else if (!height) {
    height = width
  }

  let Component = onPress || onLongPress ? TouchableOpacity : View
  if (component) {
    Component = component
  }

  return (
    <Component
      {...props}
      style={[styles.container, containerStyle && containerStyle]}>
```

Monte9 commented:
@binoy14 Thanks for your feedback. I have addressed all PR comments. And I have also added support for Icons & initials as discussed. 
Check out the updated PR description for latest avatar samples.

binoy14 commented:
@Monte9 I added the Props in the docs as we discussed. Everything lgtm and can be merged anytime
